src/nupic/algorithms/Anomaly.hpp

```diff
@@ -112,10 +113,16 @@ namespace nupic
                        const std::vector<UInt>& predicted,
                        Real64 inputValue=0, UInt timestamp=0);
 
+//helper
+        void setLikelihood(AnomalyLikelihood newInstance) {
+          this->likelihood_ = newInstance;
+        }
+
       private:
         AnomalyMode mode_;
         Real32 binaryThreshold_;
         std::unique_ptr<nupic::util::MovingAverage> movingAverage_;
+        AnomalyLikelihood likelihood_; //TODO which params/how pass them to constructor?
```

Comment:
The problem would be with different parameters for Raw/Likelihood. Or we'd have to use redundant params: `virtual Real compute(vector active, vector predicted, Real rawAnomalyScore, int timestamp)`. 
Also, I'd like to rename `anomalyProbability()` to `compute()` ,which will diverge from the python impl (we can leave comment), but align with the c++ codebase.